<a href="https://colab.research.google.com/github/yoshinaga2904/Bootcamp-Alura/blob/main/Fabio_Yoshinaga_modulo_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).




# A importância da atividade física para a saúde e seus benefícios para o cérebro.

"O tecido cerebral recebe todas as informações coletadas no meio ambiente através dos sentidos do corpo humano: visão, tato, paladar, olfato e visão. Isso faz com que o tecido cerebral realize um processo, desencadeando na movimentação de músculo. Logo, o contato do cérebro com o meio externo é dado pela da atividade física, por meio de um movimento qualquer.

O exercício físico traz inúmeros benefícios para quem o pratica, podendo ser realizado de modo profissional ou amador, considerando sempre a intensidade do movimento e respeito aos limites do corpo.

Cada pessoa irá optar pela atividade de acordo com sua finalidade. Pode ser um exercício aeróbico, de força ou equilíbrio, contanto que seja realizado de forma organizada, pois contribui com a performance do praticante.

**Importante!** Especialistas alertam que a melhor forma de obter êxito nesse processo é escolher uma atividade que agrade a pessoa que irá realizar.

Durante a realização da atividade recreativa, o cérebro é oxigenado com sangue de boa qualidade e, neste momento, muitas ações ocorrem no corpo, uma delas é a maior atividade cerebral de uma área chamada de hipocampo.

Isso resulta no melhor desempenho da memória e maior capacidade intelectual, além da liberação de endorfina, hormônio que transmite a sensação de prazer, diminui a ansiedade e ativa a serotonina, um neurotransmissor responsável por combater a depressão. "

*fonte:* https://www.educamaisbrasil.com.br/enem/educacao-fisica/a-importancia-da-atividade-fisica-para-a-saude



# Crianças e adolescentes (5 a 17 anos), incluindo aqueles com alguma “incapacidade”:

As novas recomendações da OMS para atividades físicas: o que muda 

Ao menos 60 minutos por dia de atividade física, em média, com intensidade moderada a vigorosa.

Atividades que fortalecem músculos e ossos devem ser incorporadas pelo menos 3 dias por semana.

Restrições para aqueles com “incapacidade”: devem começar com pequenas quantidades de atividade física e aumentar gradualmente a frequência, intensidade e duração. É recomendada uma consulta com um profissional de saúde especialista na condição antes do início da prática.

 

Leia mais em: https://saude.abril.com.br/fitness/as-novas-recomendacoes-da-oms-para-atividades-fisicas-o-que-muda/

# Resumo:

Analisar todas as questões respondidas pelos alunos referente à atividade física realizada. Utilizarei o questionário Pense_2015 Amostra 2, identificar o quanto os alunos praticam atividades físicas e relacionar ao estudo mencionado no artigo *A importância da atividade física para a saúde e seus benefícios para o cérebro.*

> fonte: https://www.educamaisbrasil.com.br/enem/educacao-fisica/a-importancia-da-atividade-fisica-para-a-saude.



# Conclusão:

O que fica claro com a análise dos dados da amostra Pense 2015 é que embora os dados coletados referente ao Peso, Altura e IMC dos alunos estejam aparentemente normais, os percentuais de atividades físicas realizadas estão muito abaixo do recomendado pela OMS. 

Além de ser um requisito fundamental para a saúde dos alunos, a atividade física proporciona um ganho em raciocínio e inteligência o que são comprovados por estudos.

Acreditar que impulsionando as atividades físicas dentro e fora das escolas, fará com que o alunos tenha um ganho tanto intelectual como de saúde, é um assunto que devemos considerar com a análise feita.

Então concluo que a importância da atividade física é fundamental para a saúde e seus benefícios para o cérebro.




# Biliotecas:

In [3]:
import pandas as pd
from zipfile import ZipFile
import plotly.express as px
import  plotly.graph_objects  as  go
import numpy as np
from scipy.stats import mode

# Análise do questionário Pense2015_amostra 2, Leitura do arquivo:

In [4]:
#Leitura do arquivo
caminho = "/content/drive/MyDrive/PENSE_2015/PeNSE_2015_AMOSTRA2.zip"
zip = ZipFile(caminho)
caminho_pasta_zipada = zip.namelist()[1]

dados = pd.read_csv(zip.open(caminho_pasta_zipada), sep = ";", thousands = ".", decimal = ",")
dados.head()

ANOPESQ  PAIS  REGEOGR  VB00004  VB01001  ...  V0008  V0041  aluno  escola  turma
0     2015    76        1        1        2  ...      4      1      1       1    296
1     2015    76        1        1        2  ...      4      1      2       1    296
2     2015    76        1        1        1  ...      4      1      3       1    296
3     2015    76        1        1        1  ...      4      1      4       1    296
4     2015    76        1        1        1  ...      4      1      5       1    296

[5 rows x 181 columns]

# Criando as variáveis IMC e CLASSIFICAÇÃO IMC para análises 

In [5]:
#Criando variavel IMC
dados['IMC'] = dados['VB17003']/ ((dados['VB17004']/100)**2)

#Criando variavel CLASSIFICACAO_IMC
classes = [0, 18.5, 25, 30, 60]
labels = ['ABAIXO', 'NORMAL', 'SOBREPESO', 'OBESIDADE']

dados['CLASSIFICACAO_IMC'] = pd.cut(x = dados['IMC'],
       bins = classes,
       labels = labels,
       include_lowest = True)

dados.head()

ANOPESQ  PAIS  REGEOGR  VB00004  ...  escola  turma        IMC  CLASSIFICACAO_IMC
0     2015    76        1        1  ...       1    296  15.051321             ABAIXO
1     2015    76        1        1  ...       1    296  21.727839             NORMAL
2     2015    76        1        1  ...       1    296  18.502907             NORMAL
3     2015    76        1        1  ...       1    296  23.322831             NORMAL
4     2015    76        1        1  ...       1    296  28.122076          SOBREPESO

[5 rows x 183 columns]

# Seleção das variáveis do questionário do aluno, relacionadas à Atividade Fisica 

In [6]:
#Selecionando parte do questionário para análise:
dados_pesquisa = dados[['VB03003A','VB03005A','VB03006A','VB03007','VB03011A',
                        'TEMPOEDFIS','TEMPOEXTRA','TEMPOTOTAL','TEMPOEST',
                        'VB03008','VB01001',
                        'REGEOGR','VB01003','VB17003','VB17004','IMC','CLASSIFICACAO_IMC']]
dados_pesquisa.head()                        

VB03003A  VB03005A  VB03006A  ...  VB17004        IMC  CLASSIFICACAO_IMC
0         1        -1         4  ...    157.0  15.051321             ABAIXO
1         5         8         3  ...    152.0  21.727839             NORMAL
2         3         8         8  ...    161.4  18.502907             NORMAL
3         2         1         1  ...    172.5  23.322831             NORMAL
4         3         6         2  ...    167.5  28.122076          SOBREPESO

[5 rows x 17 columns]

In [7]:
#Renomeando dados das colunas
#Dicionários
teve_aula = {1:'Nenhum dia nos últimos 7 dias (0 dia)',
             2:'1 dia nos últimos 7 dias',
             3:'2 dias nos últimos 7 dias',
             4:'3 dias nos últimos 7 dias',
             5:'4 dias nos últimos 7 dias',
             6:'5 dias nos últimos 7 dias',
             7:'5 dias mais sábado, nos últimos 7 dias',
             8:'5 dias mais sábado e domingo, nos últimos 7 dias',
             99:	'Não informado'}

quanto_tempo_por_dia = {-1:'Pulo no questionário',
                        1:'Não fiz aula de educação física na escola nos últimos 7 dias',
                        2:'Menos de 10 minutos por dia',
                        3:'10 a 19 minutos por dia',
                        4:'20 a 29 minutos por dia',
                        5:'30 a 39 minutos por dia',
                        6:'40 a 49 minutos por dia',
                        7:'50 a 59 minutos por dia',
                        8:'1 hora ou mais por dia',
                        99:'Não informado'}

quantos_dias_pratica = {1:'Nenhum dia nos últimos 7 dias (0 dia)',
                        2:'1 dia nos últimos 7 dias',
                        3:'2 dias nos últimos 7 dias',
                        4:'3 dias nos últimos 7 dias',
                        5:'4 dias nos últimos 7 dias',
                        6:'5 dias nos últimos 7 dias',
                        7:'5 dias mais sábado, nos últimos 7 dias',
                        8:'5 dias mais sábado e domingo, nos últimos 7 dias',
                        99:'Não informado'}

quanto_tempo_por_dia_duram = {-1:'Pulo no questionário',
                              1:'Menos de 10 minutos por dia',
                              2:'10 a 19 minutos por dia',
                              3:'20 a 29 minutos por dia',
                              4:'30 a 39 minutos por dia',
                              5:'40 a 49 minutos por dia',
                              6:'50 a 59 minutos por dia',
                              7:'1 hora ou mais por dia',
                              99:'Não informado'}

em_quantos_dias_voce_fez = {1:'Nenhum dia nos últimos 7 dias (0 dia)',
                            2:'1 dia nos últimos 7 dias',
                            3:'2 dias nos últimos 7 dias',
                            4:'3 dias nos últimos 7 dias',
                            5:'4 dias nos últimos 7 dias',
                            6:'5 dias nos últimos 7 dias',
                            7:'5 dias mais sábado, nos últimos 7 dias',
                            8:'5 dias mais sábado e domingo, nos últimos 7 dias',
                            99:'Não informado'}

se_voce_tivesse_oportunidade = {1:'Não faria mesmo assim',
                                2:'Faria atividade física em alguns dias da semana',
                                3:'Faria atividade física na maioria dos dias da semana',
                                4:'Já faço atividade física em alguns dias da semana',
                                5:'Já faço atividade física na maioria dos dias da semana',
                                99:'Não informado'}

sexo = {1:	'Masculino', 2:	'Feminino'}

regiao = {1:'Norte', 2:'Nordeste', 3:'Sudeste', 4:'Sul', 5:'Centro_Oeste'}

idade = {11:'11 anos ou menos',
          12:'12 anos',
          13:'13 anos',
          14:'14 anos',
          15:'15 anos',
          16:'16 anos',
          17:'17 anos',
          18:'18 anos',
          19:'19 anos ou mais'}

for coluna in dados_pesquisa[['VB03003A','VB03005A','VB03006A',
                              'VB03007','VB03011A','VB03008',
                              'VB01001','REGEOGR','VB01003']]:
  if coluna == 'VB03003A':
    dados_pesquisa['VB03003A'] = dados_pesquisa['VB03003A'].map(teve_aula)
  elif coluna == 'VB03005A': 
    dados_pesquisa['VB03005A'] = dados_pesquisa['VB03005A'].map(quanto_tempo_por_dia)
  elif coluna =='VB03006A':
    dados_pesquisa['VB03006A'] = dados_pesquisa['VB03006A'].map(quantos_dias_pratica)
  elif coluna == 'VB03007':
    dados_pesquisa['VB03007'] = dados_pesquisa['VB03007'].map(quanto_tempo_por_dia_duram)
  elif coluna == 'VB03011A':
    dados_pesquisa['VB03011A'] = dados_pesquisa['VB03011A'].map(em_quantos_dias_voce_fez)
  elif coluna == 'VB03008':
    dados_pesquisa['VB03008'] = dados_pesquisa['VB03008'].map(se_voce_tivesse_oportunidade)
  elif coluna == 'VB01001':
    dados_pesquisa['VB01001'] = dados_pesquisa['VB01001'].map(sexo)
  elif coluna == 'REGEOGR':
    dados_pesquisa['REGEOGR'] = dados_pesquisa['REGEOGR'].map(regiao)
  else:
     dados_pesquisa['VB01003'] = dados_pesquisa['VB01003'].map(idade)

dados_pesquisa.head()

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:79: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:81: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:83: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs

VB03003A  ... CLASSIFICACAO_IMC
0  Nenhum dia nos últimos 7 dias (0 dia)  ...            ABAIXO
1              4 dias nos últimos 7 dias  ...            NORMAL
2              2 dias nos últimos 7 dias  ...            NORMAL
3               1 dia nos últimos 7 dias  ...            NORMAL
4              2 dias nos últimos 7 dias  ...         SOBREPESO

[5 rows x 17 columns]

In [8]:
#renomeando colunas
questionario_resposta = dados_pesquisa
questionario_resposta.rename(columns = {'VB03003A':'TEVE_AULA', 'VB03005A':'QUANTO_TEMPO_POR_DIA',
                                        'VB03006A':'QUANTOS_DIAS_PRATICA', 'VB03007':'QUANTO_TEMPO_POR_DIA_DURAM',
                                        'VB03011A':'EM_QUANTOS_DIAS_VOCE_FEZ',
                                        'TEMPOEDFIS':'TEMPO_MEDIO_ACUMULADO','TEMPOEXTRA': 'TEMPO_MEDIO_DIARIO_ACUMULADO',
                                        'TEMPOTOTAL':'TEMPO_ACUMULADO', 'TEMPOEST':'TEMPO_GLOBAL_ESTIMADO',
                                        'VB03008':'SE_VOCE_TIVESSE_OPORTUNIDADE', 'VB01001':'SEXO',
                                        'REGEOGR':'REGIAO','VB01003':'IDADE',
                                        'VB17003':'PESO', 'VB17004':'ALTURA',
                                        'IMC':'IMC_CALCULADO', 'CLASSIFICACAO_IMC':'NIVEL_IMC'}, inplace = True)
questionario_resposta.head()

/usr/local/lib/python3.6/dist-packages/pandas/core/frame.py:4308: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



TEVE_AULA  ...  NIVEL_IMC
0  Nenhum dia nos últimos 7 dias (0 dia)  ...     ABAIXO
1              4 dias nos últimos 7 dias  ...     NORMAL
2              2 dias nos últimos 7 dias  ...     NORMAL
3               1 dia nos últimos 7 dias  ...     NORMAL
4              2 dias nos últimos 7 dias  ...  SOBREPESO

[5 rows x 17 columns]

# DataFrame Questionario Finalizado

In [9]:
#Dataframe finalizado
questionario = questionario_resposta
questionario.head()

TEVE_AULA  ...  NIVEL_IMC
0  Nenhum dia nos últimos 7 dias (0 dia)  ...     ABAIXO
1              4 dias nos últimos 7 dias  ...     NORMAL
2              2 dias nos últimos 7 dias  ...     NORMAL
3               1 dia nos últimos 7 dias  ...     NORMAL
4              2 dias nos últimos 7 dias  ...  SOBREPESO

[5 rows x 17 columns]

# Funções Gerais:

In [10]:
#DADO = Dataframe criado na função dataframe_por_questionario
#PESQUISA = dado que deseja calcular do Dataframe criado

def frequencia_e_percentual(dado,pesquisa_PERGUNTA):

  frequencia_PERGUNTA = dado[pesquisa_PERGUNTA].value_counts()
  percentual_PERGUNTA = (dado[pesquisa_PERGUNTA].value_counts(normalize = True) * 100).round(2)


  distribuicao_frequencia_PERGUNTA = pd.DataFrame({'Frequencia': frequencia_PERGUNTA, 'Percentual':percentual_PERGUNTA})
  
  return distribuicao_frequencia_PERGUNTA


In [11]:
#grafico
def grafico_frequencia(dado,pergunta_y, titulo_label_x, titulo_label_y):
  fig = px.bar(x=dado.index, 
              y=dado[pergunta_y],labels={'x':titulo_label_x,'y':titulo_label_y})

  return fig.show()


In [12]:
#medidas centrais
def medidas_centrais(dado):
  media = dado.mean()
  desvio_padrao = dado.std()
  moda = mode(dado).mode[0]
  mediana = dado.median()
  return media, desvio_padrao, moda, mediana

# **Inicio da Análise:** Aspectos Gerais dos alunos selecionados

##Sexo do aluno

Aparentemente nossa amostra está dividida igualmente entre os sexos masculino e feminino.

In [13]:
sexo = frequencia_e_percentual(questionario, 'SEXO')
sexo

Frequencia  Percentual
Masculino        8287       50.05
Feminino         8269       49.95

In [14]:
sexo_grafico = grafico_frequencia(sexo, 'Frequencia', 'Sexo', 'Quantidade de Alunos')

##Região do aluno

Aparentemente nossa amostra divide igualmente as quantidades de alunos por região

In [15]:
regiao = frequencia_e_percentual(questionario, 'REGIAO')
regiao

Frequencia  Percentual
Nordeste            3465       20.93
Centro_Oeste        3420       20.66
Sudeste             3276       19.79
Sul                 3207       19.37
Norte               3188       19.26

In [16]:
regiao_grafico = grafico_frequencia(regiao, 'Frequencia', 'Região', 'Quantidade de Alunos')

##Idade do aluno

A frequência de alunos diminue conforme a idade do aluno aumenta. O que é normal, pois quanto mais velho o aluno, significa que possivelmente ele repetiu algum ano ou entrou atrasado no ensino fundamental.


In [17]:
idade = frequencia_e_percentual(questionario, 'IDADE')
idade

Frequencia  Percentual
12 anos                 2634       15.91
13 anos                 2561       15.47
15 anos                 2425       14.65
16 anos                 2270       13.71
14 anos                 2133       12.88
11 anos ou menos        2072       12.52
17 anos                 1537        9.28
18 anos                  563        3.40
19 anos ou mais          361        2.18

In [18]:
idade_grafico = grafico_frequencia(idade, 'Frequencia', 'Idade', 'Quantidade de Alunos')

##Peso(kg) aluno

O peso médio da nossa amostra de dados é de 54.70kg, independente do sexo do aluno. O boxplot demonstra que até a idade de 13 anos, ambos os sexos mantém o mesmo peso, após essa idade o sexo masculino tende a ter maior peso em relação ao sexo feminino, o que é normal.

Com um intervalo de confiança de 95% nossa média de peso dos alunos estará em uma faixa de 54.48kg à 54.93kg.

In [19]:
fig = px.box(questionario, x="IDADE", y="PESO", color="SEXO",
             title = "Box plot para Peso por Idade",
             category_orders={"IDADE": ["11 anos ou menos",
                                        "12 anos", "13 anos",
                                        "14 anos", "15 anos",
                                        "16 anos","17 anos",
                                        "18 anos","19 anos ou mais"]},
             color_discrete_map = { "Masculino" :  "blue" , "Feminino" :  "red" })
fig.update_traces() # "exclusive" or "inclusive", or "linear" by default
fig.show()

In [20]:
media, desvio_padrao, moda, mediana = medidas_centrais(questionario['PESO'])
print(f' Média = {media}\n Desvio Padrão = {desvio_padrao}\n Moda = {moda}\n Mediana = {mediana}')

 Média = 54.70558105822677
 Desvio Padrão = 14.547864125282576
 Moda = 50.0
 Mediana = 52.9


In [21]:
#definir o gráfico histograma
fig = px.histogram(questionario, x="PESO", color="SEXO",title = "Histograma Peso",
                   color_discrete_map = { "Masculino" :  "blue" , "Feminino" :  "red" },
                   opacity=0.8, nbins=50)

#marcar posição da média
fig.add_shape(name="Média", type="line", x0=media, y0=0, x1=media, y1=3000, line=dict(color="green",width=4))

#criar anotação para identificar a média
fig.add_annotation(x=media, y=2900,text="Média",showarrow=True,arrowhead=2,ax=150)

#marcar posição da moda
fig.add_shape(name="Moda",type="line", x0=moda, y0=0, x1=moda, y1=3000, line=dict(color="orange",width=4))

#criar anotação para identificar a moda
fig.add_annotation(x=moda, y=2550,text="Moda",showarrow=True,arrowhead=2,ax=-80)

#marcar posição da mediana
fig.add_shape(name="Mediana",type="line", x0=mediana, y0=0, x1=mediana, y1=3000, line=dict(color="black",width=4))

#criar anotação para identificar a mediana
fig.add_annotation(x=mediana, y=3000,text="Mediana",showarrow=True,arrowhead=2,ax=-50)

#marcar posição da desvio padrão
fig.add_shape(name="Desvio Padrão",type="rect",x0=media-desvio_padrao, y0=0, x1=media+desvio_padrao, y1=3000,
              line=dict(color="gray",width=2,),fillcolor="gray",opacity=0.3)

#criar anotação para identificar o desvio padrão
fig.add_annotation(x=69, y=2400,text="Desvio Padrão",showarrow=True,arrowhead=2,ax=150)

#plotar linha da média, moda, mediana e desvio padrão
fig.update_shapes(dict(xref='x', yref='y'))

#criar visualização ao passar o mouse sob o gráfico
fig.update_traces(hoverinfo='x', selector=dict(type='histogram'))

#plotar visualização ao passar o mouse
fig . update_layout (hovermode = "x")

#exibir o gráfico
fig.show()

In [22]:
from statsmodels.stats.weightstats import DescrStatsW

descricao_peso = DescrStatsW(questionario["PESO"])
descricao_peso.tconfint_mean()

/usr/local/lib/python3.6/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning:

pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.



(54.48396492578986, 54.92719719066339)

O Peso médio da nossa amostra de dados respeita um intervalo de confiança de 95%, entre a faixa de 54.48 kg à 54.93 kg.  

In [23]:
from scipy.stats import normaltest

distribuicao_peso_m = questionario.query("SEXO == 'Masculino'")["PESO"]
distribuicao_peso_f = questionario.query("SEXO == 'Feminino'")["PESO"]

_, p_value_normaltest_m = normaltest(distribuicao_peso_m)
_, p_value_normaltest_f = normaltest(distribuicao_peso_f)
print(p_value_normaltest_m)
print(p_value_normaltest_f)

0.0
0.0


Em relação ao peso a distribuição por sexo (masculino e feminino) não é uma distribuição normal, pois o p_value é menor que nosso nível de significancia de 0.05, para um nivel de confiança de 95%.

In [24]:
from scipy.stats import ranksums

_, p_value_ranksums = ranksums(distribuicao_peso_m, distribuicao_peso_f)
p_value_ranksums

2.1586825995432552e-66

Pelo Peso não ser uma distribuição normal, usamos um teste não parametrico para verificar se a distribuição entre os pesos por sexo tem alguma relação entre si. Como o p_value é menor que 0.05, verificamos que as amostras para o sexo masculino e feminino são independentes uma da outra.

##Altura(cm) aluno

A altura média da nossa amostra de dados é de 160.71cm, independente do sexo do aluno. O boxplot demonstra que até a idade de 12 anos, ambos os sexos mantém o mesma altura, após essa idade o sexo masculino tende a ser maior em relação ao sexo feminino, o que é normal.

Com um intervalo de confiança de 95% nossa média de altura dos alunos estará em uma faixa de 160.55cm à 160.87cm.

In [25]:
fig = px.box(questionario, x="IDADE", y="ALTURA", color="SEXO",
             title = "Box plot para Altura por Idade",
             category_orders={"IDADE": ["11 anos ou menos",
                                        "12 anos", "13 anos",
                                        "14 anos", "15 anos",
                                        "16 anos","17 anos",
                                        "18 anos","19 anos ou mais"]},
             color_discrete_map = { "Masculino" :  "blue" , "Feminino" :  "red" })
fig.update_traces() # "exclusive" or "inclusive", or "linear" by default
fig.show()

In [26]:
media, desvio_padrao, moda, mediana = medidas_centrais(questionario['ALTURA'])
print(f' Média = {media}\n Desvio Padrão = {desvio_padrao}\n Moda = {moda}\n Mediana = {mediana}')

 Média = 160.71311307078818
 Desvio Padrão = 10.3400702338278
 Moda = 160.0
 Mediana = 160.4


In [27]:
#definir o gráfico histograma
fig = px.histogram(questionario, x="ALTURA", color="SEXO",title = "Histograma Altura",
                   color_discrete_map = { "Masculino" :  "blue" , "Feminino" :  "red" },
                   opacity=0.8, nbins=50)

#marcar posição da média
fig.add_shape(name="Média", type="line", x0=media, y0=0, x1=media, y1=1500, line=dict(color="green",width=4))

#criar anotação para identificar a média
fig.add_annotation(x=media, y=1500,text="Média",showarrow=True,arrowhead=2,ax=-25)

#marcar posição da moda
fig.add_shape(name="Moda",type="line", x0=moda, y0=0, x1=moda, y1=1500, line=dict(color="orange",width=4))

#criar anotação para identificar a moda
fig.add_annotation(x=moda, y=1500,text="Moda",showarrow=True,arrowhead=2,ax=-150)

#marcar posição da mediana
fig.add_shape(name="Mediana",type="line", x0=mediana, y0=0, x1=mediana, y1=1500, line=dict(color="black",width=4))

#criar anotação para identificar a mediana
fig.add_annotation(x=mediana, y=1500,text="Mediana",showarrow=True,arrowhead=2,ax=-75)

#marcar posição da desvio padrão
fig.add_shape(name="Desvio Padrão",type="rect",x0=media-desvio_padrao, y0=0, x1=media+desvio_padrao, y1=1500,
              line=dict(color="gray",width=2,),fillcolor="gray",opacity=0.3)

#criar anotação para identificar o desvio padrão
fig.add_annotation(x=media+desvio_padrao, y=1500,text="Desvio Padrão",showarrow=True,arrowhead=2,ax=20)

#plotar linha da média, moda, mediana e desvio padrão
fig.update_shapes(dict(xref='x', yref='y'))

#criar visualização ao passar o mouse sob o gráfico
fig.update_traces(hoverinfo='x', selector=dict(type='histogram'))

#plotar visualização ao passar o mouse
fig . update_layout (hovermode = "x")

#exibir o gráfico
fig.show()

In [28]:
from statsmodels.stats.weightstats import DescrStatsW

descricao_altura = DescrStatsW(questionario["ALTURA"])
descricao_altura.tconfint_mean()

(160.55559672649298, 160.8706294150872)

A Altura média de nossa amostra está entre 160.55cm à 160.87cm, com intervalo de confiança de 95%.

In [29]:
from scipy.stats import normaltest

distribuicao_altura_m = questionario.query("SEXO == 'Masculino'")["ALTURA"]
distribuicao_altura_f = questionario.query("SEXO == 'Feminino'")["ALTURA"]

_, p_value_normaltest_m = normaltest(distribuicao_altura_m)
_, p_value_normaltest_f = normaltest(distribuicao_altura_f)
print(p_value_normaltest_m)
print(p_value_normaltest_f)

7.695118853996524e-75
7.125189471020874e-47


A Altura embora pareça conforme o gráfico com uma distribuição normal, ela não é, nem para o sexo masculino, nem para o sexo feminino. O que nos leva a fazermos um teste não parametrico para comprovar se há relação entre as distribuições por sexo.

In [30]:
from scipy.stats import ranksums

_, p_value_ranksums = ranksums(distribuicao_altura_m, distribuicao_altura_f)
p_value_ranksums

0.0

Como o p-value é menor que o nivel de significancia do teste, 0.05, rejeitamos a hipótese que haja relação entre as alturas por sexo da nossa amostra, ou seja, elas são independentes. 

##Nivel IMC

O IMC da nossa amostra está com 55.50% em níveis considerados normais. Outros 30.05% em níveis abaixo do que é considerado normal, isso é um fator preocupante. Sobrepeso com 10.63% e obesidade com 3.82%.

O intervalo de confiança de 95% para o IMC está na faixa de 20.89 a 21.02 que é considerado um nível normal de IMC.

In [31]:
classificacao_imc = frequencia_e_percentual(questionario,'NIVEL_IMC')
classificacao_imc

Frequencia  Percentual
NORMAL           9188       55.50
ABAIXO           4975       30.05
SOBREPESO        1760       10.63
OBESIDADE         633        3.82

In [32]:
classificacao_imc_grafico = grafico_frequencia(classificacao_imc,
                                               'Frequencia',
                                               'Nivel do IMC',
                                               'Quantidade de Alunos')

In [33]:
media, desvio_padrao, moda, mediana = medidas_centrais(questionario['IMC_CALCULADO'])
print(f' Média = {media}\n Desvio Padrão = {desvio_padrao}\n Moda = {moda}\n Mediana = {mediana}')

 Média = 20.95825220663186
 Desvio Padrão = 4.244441438911331
 Moda = 16.155478395061728
 Mediana = 20.190504330368643


In [34]:
#definir o gráfico histograma
fig = px.histogram(questionario, x="IMC_CALCULADO", color="SEXO",title = "Histograma IMC por Sexo",
                   color_discrete_map = { "Masculino" :  "blue" , "Feminino" :  "red" },
                   opacity=0.7, nbins=50)

#marcar posição da média
fig.add_shape(name="Média", type="line", x0=media, y0=0, x1=media, y1=4000, line=dict(color="green",width=4))

#criar anotação para identificar a média
fig.add_annotation(x=media, y=4000,text="Média",showarrow=True,arrowhead=2,ax=-25)

#marcar posição da moda
fig.add_shape(name="Moda",type="line", x0=moda, y0=0, x1=moda, y1=4000, line=dict(color="orange",width=4))

#criar anotação para identificar a moda
fig.add_annotation(x=moda, y=4000,text="Moda",showarrow=True,arrowhead=2,ax=-150)

#marcar posição da mediana
fig.add_shape(name="Mediana",type="line", x0=mediana, y0=0, x1=mediana, y1=4000, line=dict(color="black",width=4))

#criar anotação para identificar a mediana
fig.add_annotation(x=mediana, y=4000,text="Mediana",showarrow=True,arrowhead=2,ax=-75)

#marcar posição da desvio padrão
fig.add_shape(name="Desvio Padrão",type="rect",x0=media-desvio_padrao, y0=0, x1=media+desvio_padrao, y1=4000,
              line=dict(color="gray",width=2,),fillcolor="gray",opacity=0.3)

#criar anotação para identificar o desvio padrão
fig.add_annotation(x=media+desvio_padrao, y=4000,text="Desvio Padrão",showarrow=True,arrowhead=2,ax=20)

#plotar linha da média, moda, mediana e desvio padrão
fig.update_shapes(dict(xref='x', yref='y'))

#criar visualização ao passar o mouse sob o gráfico
fig.update_traces(hoverinfo='x', selector=dict(type='histogram'))

#plotar visualização ao passar o mouse
fig . update_layout (hovermode = "x")

#exibir o gráfico
fig.show()

In [35]:
from statsmodels.stats.weightstats import DescrStatsW

descricao_imc = DescrStatsW(questionario["IMC_CALCULADO"])
descricao_imc.tconfint_mean()

(20.893594144943506, 21.022910268320153)

O intervalo de confiança de 95% para o IMC está na faixa de 20.89 a 21.02 que é considerado um nível normal de IMC.

In [36]:
from scipy.stats import normaltest

distribuicao_imc_m = questionario.query("SEXO == 'Masculino'")["IMC_CALCULADO"]
distribuicao_imc_f = questionario.query("SEXO == 'Feminino'")["IMC_CALCULADO"]

_, p_value_normaltest_m = normaltest(distribuicao_imc_m)
_, p_value_normaltest_f = normaltest(distribuicao_imc_f)
print(p_value_normaltest_m)
print(p_value_normaltest_f)

0.0
0.0


O IMC entre os sexo apresentam um p_value inferior a 0.05, o que nos classifica como não sendo uma distribuição normal. Faremos um teste não parametrico para comprovar se há relação entre os sexos.

In [37]:
from scipy.stats import ranksums

_, p_value_ranksums = ranksums(distribuicao_imc_m, distribuicao_imc_f)
p_value_ranksums

2.8089346699927397e-08

Como o p-value é menor que o nivel de significancia do teste, 0.05, rejeitamos a hipótese que haja relação entre os IMC's por sexo da nossa amostra, ou seja, elas são independentes.

# **Questão 1** - NOS ÚLTIMOS 7 DIAS, quantos dias você teve aulas de educação física na escola?

Segundo a OMS atividades fisicas devem ser incorporadas em pelo mesmo 3 dias por semana, ou seja, nossa amostra está abaixo dos níveis considerados pela OMS em 88,59%.

In [38]:
#Calculo frequência e percentual
teve_aula = frequencia_e_percentual(questionario,'TEVE_AULA')
teve_aula

Frequencia  Percentual
1 dia nos últimos 7 dias                                6569       39.68
2 dias nos últimos 7 dias                               4858       29.34
Nenhum dia nos últimos 7 dias (0 dia)                   3240       19.57
3 dias nos últimos 7 dias                                982        5.93
5 dias nos últimos 7 dias                                348        2.10
4 dias nos últimos 7 dias                                261        1.58
5 dias mais sábado e domingo, nos últimos 7 dias         139        0.84
5 dias mais sábado, nos últimos 7 dias                   107        0.65
Não informado                                             52        0.31

In [39]:
#gráfico
teve_aulas_grafico = grafico_frequencia(teve_aula, 'Frequencia',
                                                   'Teve Atividade Fisica?', 
                                                   'Quantidade de Alunos')

# **Questão 2** - NOS ÚLTIMOS 7 DIAS, quanto tempo por dia você fez atividade física ou esporte durante as aulas de educação física na escola?

Pela OMS o ideal seria ao menos 60 minutos por dia de atividade física, em média, com intensidade moderada a vigorosa. Considerando que 19.88% não responderam ao questionário, apenas 15.47% praticam 1 hora ou mais por dia. 

In [40]:
quanto_tempo_por_dia = frequencia_e_percentual(questionario, 'QUANTO_TEMPO_POR_DIA')
quanto_tempo_por_dia

Frequencia  Percentual
Pulo no questionário                                      3292       19.88
1 hora ou mais por dia                                    2562       15.47
40 a 49 minutos por dia                                   2320       14.01
Não fiz aula de educação física na escola nos ú...        1905       11.51
30 a 39 minutos por dia                                   1503        9.08
50 a 59 minutos por dia                                   1318        7.96
10 a 19 minutos por dia                                   1255        7.58
Menos de 10 minutos por dia                               1226        7.41
20 a 29 minutos por dia                                   1143        6.90
Não informado                                               32        0.19

In [41]:
quanto_tempo_por_dia_grafico = grafico_frequencia(quanto_tempo_por_dia,
                                      'Frequencia',
                                      'Quanto tempo por dia fez Atividade Fisica?',
                                      'Quantidade de Alunos')

# **Questão 3** - NOS ÚLTIMOS 7 DIAS, sem contar as aulas de educação física da escola, em quantos dias você praticou alguma atividade física, como esportes, dança, ginástica, musculação, lutas ou outra atividade?

Considerando que a OMS estipula que ao menos seja praticado 3 dias por semana, 58.21% dos alunos não seguem estas recomendações de praticarem atividades fisicas.

In [42]:
quanto_dias_pratica = frequencia_e_percentual(questionario, 'QUANTOS_DIAS_PRATICA')
quanto_dias_pratica

Frequencia  Percentual
Nenhum dia nos últimos 7 dias (0 dia)                   5507       33.26
2 dias nos últimos 7 dias                               2092       12.64
1 dia nos últimos 7 dias                                2038       12.31
5 dias mais sábado e domingo, nos últimos 7 dias        1925       11.63
3 dias nos últimos 7 dias                               1696       10.24
5 dias nos últimos 7 dias                               1420        8.58
4 dias nos últimos 7 dias                               1054        6.37
5 dias mais sábado, nos últimos 7 dias                   767        4.63
Não informado                                             57        0.34

In [43]:
quanto_dias_pratica_grafico = grafico_frequencia(quanto_dias_pratica,'Frequencia',
                                                 'Quantos dias você praticou alguma atividade física',
                                                 'Quantidade de Alunos')

# **Questão 4** - NORMALMENTE, quanto tempo por dia duram essas atividades (como esportes, dança, ginástica, musculação, lutas ou outra atividade) que você faz? (Sem contar as aulas de educação física)

Considerando quem pulou a resposta no questionário 33.61% apenas 28.01% estão dentro das recomendações da OMS para atividade fisica de pelo menos 60 minutos por dia.

In [44]:
quanto_tempo_por_dia_duram = frequencia_e_percentual(questionario, 'QUANTO_TEMPO_POR_DIA_DURAM')
quanto_tempo_por_dia_duram

Frequencia  Percentual
Pulo no questionário               5564       33.61
1 hora ou mais por dia             4638       28.01
30 a 39 minutos por dia            1205        7.28
10 a 19 minutos por dia            1187        7.17
Menos de 10 minutos por dia        1149        6.94
20 a 29 minutos por dia            1093        6.60
40 a 49 minutos por dia             863        5.21
50 a 59 minutos por dia             821        4.96
Não informado                        36        0.22

In [45]:
quanto_tempo_por_dia_duram_grafico = grafico_frequencia(quanto_tempo_por_dia_duram,
                                                        'Frequencia',
                                                        'Quanto tempo por dia duram essas atividades',
                                                        'Quantidade de Alunos')

# **Questão 5** - NOS ÚLTIMOS 7 DIAS, em quantos dias você fez atividade física por pelo menos 60 minutos (1 hora) por dia? (Some todo o tempo que você gastou em qualquer tipo de atividade física EM CADA DIA)

Nesta questão fica muito claro o quanto é baixo o percentual que praticam ao menos 60 minutos por dia de atividade física, em média, com intensidade moderada a vigorosa. Somente 8.44%

In [46]:
em_quantos_dias_voce_fez = frequencia_e_percentual(questionario, 'EM_QUANTOS_DIAS_VOCE_FEZ')
em_quantos_dias_voce_fez

Frequencia  Percentual
Nenhum dia nos últimos 7 dias (0 dia)                   5544       33.49
1 dia nos últimos 7 dias                                2786       16.83
2 dias nos últimos 7 dias                               2046       12.36
3 dias nos últimos 7 dias                               1690       10.21
5 dias mais sábado e domingo, nos últimos 7 dias        1398        8.44
5 dias nos últimos 7 dias                               1239        7.48
4 dias nos últimos 7 dias                                947        5.72
5 dias mais sábado, nos últimos 7 dias                   796        4.81
Não informado                                            110        0.66

In [47]:
em_quantos_dias_voce_fez_grafico = grafico_frequencia(em_quantos_dias_voce_fez,
                                                      'Frequencia',
                                                      'quantos dias você fez atividade física por pelo menos 60 minutos (1 hora) por dia',
                                                      'Quantidade de Alunos')

# **Questão 6** - Se você tivesse oportunidade de fazer atividade física na maioria dos dias da semana, qual seria a sua atitude?

Aqui um dado interessante, cerca de 34,04% dizem fazerem atividades fisicas em alguns ou na maioria dos dias. Outros 60.35% dizem que fariam atividades fisicas em alguns ou na maioria dos dias, caso houvesse disponibilidade. Existe uma esperança que fariam atividades fisicas. 

In [48]:
se_voce_tivesse_oportunidade = frequencia_e_percentual(questionario, 'SE_VOCE_TIVESSE_OPORTUNIDADE')
se_voce_tivesse_oportunidade

Frequencia  Percentual
Faria atividade física na maioria dos dias da s...        5051       30.51
Faria atividade física em alguns dias da semana           4940       29.84
Já faço atividade física na maioria dos dias da...        3373       20.37
Já faço atividade física em alguns dias da semana         2264       13.67
Não faria mesmo assim                                      878        5.30
Não informado                                               50        0.30

In [49]:
se_voce_tivesse_oportunidade_grafico = grafico_frequencia(se_voce_tivesse_oportunidade,
                                                          'Frequencia',
                                                          'Se você tivesse oportunidade de fazer atividade física',
                                                          'Quantidade de Alunos')

# **Questão 7** - Refere-se ao tempo médio acumulado, nos últimos sete dias anteriores à pesquisa, que o escolar fez atividade física ou esporte durante as aulas de educação física na escola. Em minutos.

Se compararmos o tempo médio por semana, considerando o recomendado pela OMS que é ao menos 60 minutos diários, teriamos que ter um tempo semanal de 420 minutos. Assim fica muito claro o quanto estamos abaixo do recomendado pela OMS nesta questão, com apenas 0.37% da amostra que fazem o recomendado de atividade fisica

In [50]:
tamanho_amostra = len(questionario)

k = int((1 + ((10/3)*np.log10(tamanho_amostra))))
questionario["CLASSE_TEMPO_MEDIO_ACUMULADO"] = pd.cut(x = questionario["TEMPO_MEDIO_ACUMULADO"],
       bins = k,
       include_lowest = True)

frequencia = questionario["CLASSE_TEMPO_MEDIO_ACUMULADO"].value_counts()
percentual = questionario["CLASSE_TEMPO_MEDIO_ACUMULADO"].value_counts(normalize = True) * 100

distribuicao_frequencia_tempo_medio = pd.DataFrame({"Frequencia": frequencia, "Percentual": percentual })
distribuicao_frequencia_tempo_medio

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:6: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



Frequencia  Percentual
(-0.456, 30.333]          7959   48.318358
(60.667, 91.0]            3220   19.548324
(30.333, 60.667]          2775   16.846770
(121.333, 151.667]        1049    6.368383
(91.0, 121.333]            750    4.553181
(182.0, 212.333]           179    1.086693
(151.667, 182.0]           174    1.056338
(303.333, 333.667]          94    0.570665
(242.667, 273.0]            75    0.455318
(424.667, 455.0]            62    0.376396
(212.333, 242.667]          59    0.358184
(364.0, 394.333]            40    0.242836
(273.0, 303.333]            36    0.218553
(394.333, 424.667]           0    0.000000
(333.667, 364.0]             0    0.000000

In [51]:
distribuicao_frequencia_tempo_medio = distribuicao_frequencia_tempo_medio.reset_index()
distribuicao_frequencia_tempo_medio = distribuicao_frequencia_tempo_medio.rename(index={0:'0 a 30 minutos',
                                                                                        1:'60 a 90 minutos',
                                                                                        2:'30 a 60 minutos',
                                                                                        3:'120 a 150 minutos',
                                                                                        4:'90 a 120 minutos',
                                                                                        5:'180 a 210 minutos',
                                                                                        6:'150 a 180 minutos',
                                                                                        7:'300 a 330 minutos',
                                                                                        8:'240 a 270 minutos',
                                                                                        9:'420 a 450 minutos',
                                                                                        10:'210 a 240 minutos',
                                                                                        11:'360 a 390 minutos',
                                                                                        12:'270 a 300 minutos',
                                                                                        13:'390 a 420 minutos',
                                                                                        14:'330 a 360 minutos'})
distribuicao_frequencia_tempo_medio = distribuicao_frequencia_tempo_medio.rename(columns={'index':'Classes'})
distribuicao_frequencia_tempo_medio

Classes  Frequencia  Percentual
0 a 30 minutos       (-0.456, 30.333]        7959   48.318358
60 a 90 minutos        (60.667, 91.0]        3220   19.548324
30 a 60 minutos      (30.333, 60.667]        2775   16.846770
120 a 150 minutos  (121.333, 151.667]        1049    6.368383
90 a 120 minutos      (91.0, 121.333]         750    4.553181
180 a 210 minutos    (182.0, 212.333]         179    1.086693
150 a 180 minutos    (151.667, 182.0]         174    1.056338
300 a 330 minutos  (303.333, 333.667]          94    0.570665
240 a 270 minutos    (242.667, 273.0]          75    0.455318
420 a 450 minutos    (424.667, 455.0]          62    0.376396
210 a 240 minutos  (212.333, 242.667]          59    0.358184
360 a 390 minutos    (364.0, 394.333]          40    0.242836
270 a 300 minutos    (273.0, 303.333]          36    0.218553
390 a 420 minutos  (394.333, 424.667]           0    0.000000
330 a 360 minutos    (333.667, 364.0]           0    0.000000

In [52]:
fig = px.bar(distribuicao_frequencia_tempo_medio, x=distribuicao_frequencia_tempo_medio.index,
             y='Frequencia',title = "Tempo médio acumulado que o escolar fez atividade física (Periodo 7 dias anteriores)",
             labels={'x':'Tempo em aulas de educação física na escola'})
fig.show()

# **Questão 8** - Refere-se ao tempo médio diário acumulado pelo escolar com a prática de alguma atividade física extraescolar como esportes, dança, ginástica, musculação, lutas ou outra atividade, nos últimos sete dias anteriores à data da pesquisa. Em minutos

Considerando o tempo médio diário acumulado temos que apenas 7.55% da nossa amostra estão dento do recomendado pela OMS.

In [53]:
tamanho_amostra = len(questionario)

k = int((1 + ((10/3)*np.log10(tamanho_amostra))))
questionario["CLASSE_TEMPO_MEDIO_DIARIO_ACUMULADO"] = pd.cut(x = questionario["TEMPO_MEDIO_DIARIO_ACUMULADO"],
       bins = k,
       include_lowest = True)

frequencia = questionario["CLASSE_TEMPO_MEDIO_DIARIO_ACUMULADO"].value_counts()
percentual = questionario["CLASSE_TEMPO_MEDIO_DIARIO_ACUMULADO"].value_counts(normalize = True) * 100

distribuicao_frequencia_tempo_medio_diario = pd.DataFrame({"Frequencia": frequencia, "Percentual": percentual })
distribuicao_frequencia_tempo_medio_diario

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:6: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



Frequencia  Percentual
(-0.456, 30.333]          7434   45.155804
(60.667, 91.0]            1256    7.629229
(424.667, 455.0]          1243    7.550264
(121.333, 151.667]        1241    7.538116
(30.333, 60.667]          1090    6.620907
(303.333, 333.667]         841    5.108425
(182.0, 212.333]           708    4.300553
(242.667, 273.0]           639    3.881431
(91.0, 121.333]            626    3.802466
(364.0, 394.333]           581    3.529126
(151.667, 182.0]           463    2.812367
(212.333, 242.667]         216    1.312033
(273.0, 303.333]           125    0.759278
(394.333, 424.667]           0    0.000000
(333.667, 364.0]             0    0.000000

In [54]:
distribuicao_frequencia_tempo_medio_diario = distribuicao_frequencia_tempo_medio_diario.reset_index()
distribuicao_frequencia_tempo_medio_diario = distribuicao_frequencia_tempo_medio_diario.rename(index={0:'0 a 30 minutos',
                                                                                                      1:'60 a 90 minutos',
                                                                                                      2:'420 a 450 minutos',
                                                                                                      3:'120 a 150 minutos',
                                                                                                      4:'30 a 60 minutos',
                                                                                                      5:'300 a 330 minutos',
                                                                                                      6:'180 a 210 minutos',
                                                                                                      7:'240 a 270 minutos',
                                                                                                      8:'90 a 120 minutos',
                                                                                                      9:'360 a 390 minutos',
                                                                                                      10:'150 a 180 minutos',
                                                                                                      11:'210 a 240 minutos',
                                                                                                      12:'270 a 300 minutos',
                                                                                                      13:'390 a 420 minutos',   
                                                                                                      14:'330 a 360 minutos'})
distribuicao_frequencia_tempo_medio_diario = distribuicao_frequencia_tempo_medio_diario.rename(columns={'index':'Classes'})
distribuicao_frequencia_tempo_medio_diario

Classes  Frequencia  Percentual
0 a 30 minutos       (-0.456, 30.333]        7434   45.155804
60 a 90 minutos        (60.667, 91.0]        1256    7.629229
420 a 450 minutos    (424.667, 455.0]        1243    7.550264
120 a 150 minutos  (121.333, 151.667]        1241    7.538116
30 a 60 minutos      (30.333, 60.667]        1090    6.620907
300 a 330 minutos  (303.333, 333.667]         841    5.108425
180 a 210 minutos    (182.0, 212.333]         708    4.300553
240 a 270 minutos    (242.667, 273.0]         639    3.881431
90 a 120 minutos      (91.0, 121.333]         626    3.802466
360 a 390 minutos    (364.0, 394.333]         581    3.529126
150 a 180 minutos    (151.667, 182.0]         463    2.812367
210 a 240 minutos  (212.333, 242.667]         216    1.312033
270 a 300 minutos    (273.0, 303.333]         125    0.759278
390 a 420 minutos  (394.333, 424.667]           0    0.000000
330 a 360 minutos    (333.667, 364.0]           0    0.000000

In [55]:
fig = px.bar(distribuicao_frequencia_tempo_medio_diario, x=distribuicao_frequencia_tempo_medio_diario.index,
             y='Frequencia',title = "Tempo médio diário acumulado pelo escolar com a prática de alguma atividade física (Periodo 7 dias anteriores)",
             labels={'x':'Atividade física extraescolar como esportes, dança, ginástica, musculação, lutas ou outra atividade'})
fig.show()

# **Questão 9** - A atividade física acumulada foi estimada calculando o produto entre o número de dias e o tempo médio que os escolares gastam em atividades físicas, nos sete dias anteriores à pesquisa, considerando os seguintes domínios: ir e voltar da escola, aulas de educação física e outras atividades extraescolares. Em minutos.

In [56]:
tamanho_amostra = len(questionario)

k = int((1 + ((10/3)*np.log10(tamanho_amostra))))
questionario["CLASSE_TEMPO_ACUMULADO"] = pd.cut(x = questionario["TEMPO_ACUMULADO"],
       bins = k,
       include_lowest = True)

frequencia = questionario["CLASSE_TEMPO_ACUMULADO"].value_counts()
percentual = questionario["CLASSE_TEMPO_ACUMULADO"].value_counts(normalize = True) * 100

distribuicao_frequencia_tempo_medio_total_acumulado = pd.DataFrame({"Frequencia": frequencia, "Percentual": percentual })
distribuicao_frequencia_tempo_medio_total_acumulado

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:6: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



Frequencia  Percentual
(-1.821, 121.333]           5897   35.661587
(121.333, 242.667]          3896   23.560716
(242.667, 364.0]            2608   15.771650
(364.0, 485.333]            1854   11.211901
(485.333, 606.667]          1317    7.964441
(606.667, 728.0]             534    3.229318
(728.0, 849.333]             245    1.481616
(849.333, 970.667]            93    0.562409
(970.667, 1092.0]             38    0.229802
(1092.0, 1213.333]            20    0.120948
(1334.667, 1456.0]            12    0.072569
(1213.333, 1334.667]          11    0.066522
(1698.667, 1820.0]             5    0.030237
(1577.333, 1698.667]           3    0.018142
(1456.0, 1577.333]             3    0.018142

In [57]:
distribuicao_frequencia_tempo_medio_total_acumulado = distribuicao_frequencia_tempo_medio_total_acumulado.reset_index()
distribuicao_frequencia_tempo_medio_total_acumulado = distribuicao_frequencia_tempo_medio_total_acumulado.rename(index={0:'0 a 120 minutos',
                                                                                                                        1:'120 a 240 minutos',
                                                                                                                        2:'240 a 360 minutos',
                                                                                                                        3:'360 a 480 minutos',
                                                                                                                        4:'480 a 600 minutos',
                                                                                                                        5:'600 a 720 minutos',
                                                                                                                        6:'720 a 850 minutos',
                                                                                                                        7:'850 a 970 minutos',
                                                                                                                        8:'970 a 1010 minutos',
                                                                                                                        9:'1010 a 1210 minutos',
                                                                                                                        10:'1330 a 1450 minutos',
                                                                                                                        11:'1210 a 1330 minutos',
                                                                                                                        12:'1700 a 1820 minutos',
                                                                                                                        13:'1580 a 1700 minutos',
                                                                                                                        14:'1450 a 1580 minutos'})
distribuicao_frequencia_tempo_medio_total_acumulado = distribuicao_frequencia_tempo_medio_total_acumulado.rename(columns={'index':'Classes'})
distribuicao_frequencia_tempo_medio_total_acumulado

Classes  Frequencia  Percentual
0 a 120 minutos         (-1.821, 121.333]        5897   35.661587
120 a 240 minutos      (121.333, 242.667]        3896   23.560716
240 a 360 minutos        (242.667, 364.0]        2608   15.771650
360 a 480 minutos        (364.0, 485.333]        1854   11.211901
480 a 600 minutos      (485.333, 606.667]        1317    7.964441
600 a 720 minutos        (606.667, 728.0]         534    3.229318
720 a 850 minutos        (728.0, 849.333]         245    1.481616
850 a 970 minutos      (849.333, 970.667]          93    0.562409
970 a 1010 minutos      (970.667, 1092.0]          38    0.229802
1010 a 1210 minutos    (1092.0, 1213.333]          20    0.120948
1330 a 1450 minutos    (1334.667, 1456.0]          12    0.072569
1210 a 1330 minutos  (1213.333, 1334.667]          11    0.066522
1700 a 1820 minutos    (1698.667, 1820.0]           5    0.030237
1580 a 1700 minutos  (1577.333, 1698.667]           3    0.018142
1450 a 1580 minutos    (1456.0, 1577.333]           3    0.018142

In [58]:
fig = px.bar(distribuicao_frequencia_tempo_medio_total_acumulado, x=distribuicao_frequencia_tempo_medio_total_acumulado.index,
             y='Frequencia',title = "Atividade física acumulada - tempo médio que os escolares gastam em atividades físicas (Periodo 7 dias anteriores)",
             labels={'x':'Atividade física ir e voltar da escola, aulas de educação física e outras atividades extraescolares'})
fig.show()

# **Questão 10** - A atividade física globalmente estimada refere-se ao número de dias que os escolares declararam fazer, pelo menos, uma hora por dia de atividade física, nos sete dias anteriores à pesquisa. Em minutos.

In [59]:
tamanho_amostra = len(questionario)

k = int((1 + ((10/3)*np.log10(tamanho_amostra))))
questionario["CLASSE_TEMPO_GLOBAL_ESTIMADO"] = pd.cut(x = questionario["TEMPO_GLOBAL_ESTIMADO"],
       bins = k,
       include_lowest = True)

frequencia = questionario["CLASSE_TEMPO_GLOBAL_ESTIMADO"].value_counts()
percentual = questionario["CLASSE_TEMPO_GLOBAL_ESTIMADO"].value_counts(normalize = True) * 100

distribuicao_frequencia_tempo_global_estimado = pd.DataFrame({"Frequencia": frequencia, "Percentual": percentual })
distribuicao_frequencia_tempo_global_estimado

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:6: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



Frequencia  Percentual
(-0.421, 28.0]        5544   33.710325
(56.0, 84.0]          2786   16.940289
(112.0, 140.0]        2046   12.440715
(168.0, 196.0]        1690   10.276055
(392.0, 420.0]        1398    8.500547
(280.0, 308.0]        1239    7.533747
(224.0, 252.0]         947    5.758239
(336.0, 364.0]         796    4.840083
(364.0, 392.0]           0    0.000000
(308.0, 336.0]           0    0.000000
(252.0, 280.0]           0    0.000000
(196.0, 224.0]           0    0.000000
(140.0, 168.0]           0    0.000000
(84.0, 112.0]            0    0.000000
(28.0, 56.0]             0    0.000000

In [60]:
distribuicao_frequencia_tempo_global_estimado.index

CategoricalIndex([(-0.421, 28.0],   (56.0, 84.0], (112.0, 140.0],
                  (168.0, 196.0], (392.0, 420.0], (280.0, 308.0],
                  (224.0, 252.0], (336.0, 364.0], (364.0, 392.0],
                  (308.0, 336.0], (252.0, 280.0], (196.0, 224.0],
                  (140.0, 168.0],  (84.0, 112.0],   (28.0, 56.0]],
                 categories=[(-0.421, 28.0], (28.0, 56.0], (56.0, 84.0], (84.0, 112.0], (112.0, 140.0], (140.0, 168.0], (168.0, 196.0], (196.0, 224.0], ...], ordered=True, dtype='category')

In [61]:
distribuicao_frequencia_tempo_global_estimado = distribuicao_frequencia_tempo_global_estimado.reset_index()
distribuicao_frequencia_tempo_global_estimado = distribuicao_frequencia_tempo_global_estimado.rename(index={0:'0 a 28 minutos',
                                                                                                            1:'56 a 84 minutos',
                                                                                                            2:'112 a 140 minutos',
                                                                                                            3:'168 a 196 minutos',
                                                                                                            4:'392 a 420 minutos',
                                                                                                            5:'280 a 308 minutos',
                                                                                                            6:'224 a 252 minutos',
                                                                                                            7:'336 a 364 minutos',
                                                                                                            8:'364 a 392 minutos',
                                                                                                            9:'308 a 336 minutos',
                                                                                                            10:'252 a 280 minutos',
                                                                                                            11:'196 a 224 minutos',
                                                                                                            12:'140 a 168 minutos',
                                                                                                            13:'84 a 112 minutos',
                                                                                                            14:'28 a 56 minutos'})
distribuicao_frequencia_tempo_global_estimado = distribuicao_frequencia_tempo_global_estimado.rename(columns={'index':'Classes'})
distribuicao_frequencia_tempo_global_estimado

Classes  Frequencia  Percentual
0 a 28 minutos     (-0.421, 28.0]        5544   33.710325
56 a 84 minutos      (56.0, 84.0]        2786   16.940289
112 a 140 minutos  (112.0, 140.0]        2046   12.440715
168 a 196 minutos  (168.0, 196.0]        1690   10.276055
392 a 420 minutos  (392.0, 420.0]        1398    8.500547
280 a 308 minutos  (280.0, 308.0]        1239    7.533747
224 a 252 minutos  (224.0, 252.0]         947    5.758239
336 a 364 minutos  (336.0, 364.0]         796    4.840083
364 a 392 minutos  (364.0, 392.0]           0    0.000000
308 a 336 minutos  (308.0, 336.0]           0    0.000000
252 a 280 minutos  (252.0, 280.0]           0    0.000000
196 a 224 minutos  (196.0, 224.0]           0    0.000000
140 a 168 minutos  (140.0, 168.0]           0    0.000000
84 a 112 minutos    (84.0, 112.0]           0    0.000000
28 a 56 minutos      (28.0, 56.0]           0    0.000000

In [62]:
fig = px.bar(distribuicao_frequencia_tempo_global_estimado, x=distribuicao_frequencia_tempo_global_estimado.index,
             y='Frequencia',title = "A atividade física globalmente estimada (Periodo 7 dias anteriores)",
             labels={'x':'escolares declararam fazer, pelo menos, uma hora por dia de atividade física'})
fig.show()